src/domain/usecases/GetAllBasicDataSetsInfoUseCase.ts

```diff
@@ -2,16 +2,14 @@ import { FutureData } from "$/data/api-futures";
 import { BasicDataSet } from "$/domain/entities/BasicDataSet";
 import { OrgUnit } from "$/domain/entities/OrgUnit";
 import { Repositories } from "$/CompositionRoot";
-import _c from "$/domain/entities/generic/Collection";
+import _ from "$/domain/entities/generic/Collection";
+import { getId } from "$/domain/entities/Ref";
 
 export class GetAllBasicDataSetsInfoUseCase {
     constructor(private repositories: Repositories) {}
 
     public execute(orgUnits: OrgUnit[]): FutureData<BasicDataSet[]> {
-        if (_c(orgUnits).isEmpty()) {
-            return this.repositories.dataSetRepository.getBasic([]);
-        } else {
-            return this.repositories.dataSetRepository.getBasic(orgUnits.map(({ id }) => id));
-        }
+        if (_(orgUnits).isEmpty()) return this.repositories.dataSetRepository.getBasic([]);
+        return this.repositories.dataSetRepository.getBasic(orgUnits.map(getId));
```

Comment:
I meant, why the branching? the last line is enough. It does not change anything if it's empty or not.